src/cephadm/cephadm.py

```diff
@@ -4818,15 +4828,22 @@ def command_agent(ctx: CephadmContext) -> None:
 
 ##################################
 
-
-@infer_image
+@executes_early
 def command_version(ctx):
     # type: (CephadmContext) -> int
-    c = CephContainer(ctx, ctx.image, 'ceph', ['--version'])
-    out, err, ret = call(ctx, c.run_cmd(), desc=c.entrypoint)
-    if not ret:
-        print(out.strip())
-    return ret
+    import importlib
+
+    try:
+        vmod = importlib.import_module('_version')
+    except ImportError:
+        print('cephadm version UNKNOWN')
+        return 1
+    print('cephadm version {0} ({1}) {2} ({3})'.format(
+        vmod.CEPH_GIT_NICE_VER,
```

Comment:
I guess we actually _do_ run `cephadm version` in the ci testing somewhere.
```
2023-05-03T13:58:16.397 INFO:tasks.workunit.client.0.smithi162.stderr:+ sudo CEPHADM_IMAGE=quay.ceph.io/ceph-ci/ceph:pacific tmp.test_cephadm.sh.XUpEHR/tmp.cephadm.z5Br0q version
2023-05-03T13:58:16.440 INFO:tasks.workunit.client.0.smithi162.stderr:Traceback (most recent call last):
2023-05-03T13:58:16.441 INFO:tasks.workunit.client.0.smithi162.stderr:  File "/usr/lib/python3.8/runpy.py", line 194, in _run_module_as_main
2023-05-03T13:58:16.441 INFO:tasks.workunit.client.0.smithi162.stderr:    return _run_code(code, main_globals, None,
2023-05-03T13:58:16.441 INFO:tasks.workunit.client.0.smithi162.stderr:  File "/usr/lib/python3.8/runpy.py", line 87, in _run_code
2023-05-03T13:58:16.441 INFO:tasks.workunit.client.0.smithi162.stderr:    exec(code, run_globals)
2023-05-03T13:58:16.441 INFO:tasks.workunit.client.0.smithi162.stderr:  File "tmp.test_cephadm.sh.XUpEHR/tmp.cephadm.z5Br0q/__main__.py", line 10142, in <module>
2023-05-03T13:58:16.441 INFO:tasks.workunit.client.0.smithi162.stderr:  File "tmp.test_cephadm.sh.XUpEHR/tmp.cephadm.z5Br0q/__main__.py", line 10108, in main
2023-05-03T13:58:16.441 INFO:tasks.workunit.client.0.smithi162.stderr:  File "tmp.test_cephadm.sh.XUpEHR/tmp.cephadm.z5Br0q/__main__.py", line 4858, in command_version
2023-05-03T13:58:16.442 INFO:tasks.workunit.client.0.smithi162.stderr:AttributeError: module '_version' has no attribute 'CEPH_GIT_NICE_VER'
```
Will need to look a bit closer at what we're doing here, as the logs seems to be referencing pacific.
https://pulpito.ceph.com/adking-2023-05-03_12:22:42-orch:cephadm-wip-adk-testing-2023-05-02-1906-distro-default-smithi/7260944/

Reply:
the cephadm workunit has a version test. See also https://github.com/ceph/ceph/commit/1dccdd2fa182